src/main/java/com/spotinst/sdkjava/model/service/azure/statefulNode/SpotinstAzureStatefulNodeService.java

```diff
@@ -0,0 +1,310 @@
+        }
+        return retVal;
+    }
+
+    public static Boolean deleteNode(ApiStatefulNodeDeallocationConfig nodeToDelete, String nodeId , String authToken, String account) throws SpotinstHttpException {
+        // Init retVal
+        Boolean retVal = null;
+
+        // Get endpoint
+        SpotinstHttpConfig config      = SpotinstHttpContext.getInstance().getConfiguration();
+        String             apiEndpoint = config.getEndpoint();
+
+        Map<String, String> queryParams = new HashMap<String, String>();
+
+        // Add account Id Query param
+        if (account != null) {
+            queryParams.put("accountId", account);
+        }
+
+        // Get the headers
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Write to json
+        Map<String, ApiStatefulNodeDeallocationConfig> nodeToDeleteRequest = new HashMap<>();
+        nodeToDeleteRequest.put("deallocationConfig", nodeToDelete);
+        String body = JsonMapper.toJson(nodeToDeleteRequest);
+
+        // Build URI
+        String uri = String.format("%s/azure/compute/statefulNode/%s", apiEndpoint, nodeId);
+
+        // Send the request
+        RestResponse response = RestClient.sendDelete(uri, body, headers, queryParams);
+
+        // Handle the response.
+        BaseServiceEmptyResponse emptyResponse = getCastedResponse(response, BaseServiceEmptyResponse.class);
+        if (emptyResponse.getResponse().getStatus().getCode() == HttpStatus.SC_OK) {
+            retVal = true;
+        }
+
+        return retVal;
+    }
+
+    public static List<ApiStatefulNode> getAllNodes(String authToken, String account) throws SpotinstHttpException {
+        // Init retVal
+        List<ApiStatefulNode> retVal = new LinkedList<>();
+
+        // Get endpoint
+        SpotinstHttpConfig config      = SpotinstHttpContext.getInstance().getConfiguration();
+        String             apiEndpoint = config.getEndpoint();
+
+        Map<String, String> queryParams = new HashMap<String, String>();
+
+        // Add account Id Query param
+        if (account != null) {
+            queryParams.put("accountId", account);
+        }
+
+        // Get the headers
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/azure/compute/statefulNode", apiEndpoint);
+
+        // Send the request
+        RestResponse response = RestClient.sendGet(uri, headers, queryParams);
+
+        // Handle the response.
+        StatefulNodeApiResponse statefulNodeApiResponse = getCastedResponse(response, StatefulNodeApiResponse.class);
+
+        if (statefulNodeApiResponse.getResponse().getItems().size() > 0) {
+            retVal = statefulNodeApiResponse.getResponse().getItems();
+        }
+
+        return retVal;
+    }
+
+    public static Boolean recycleNode(StatefulNodeStateRequest recycleNodeRequest, String nodeId , String authToken, String account) throws SpotinstHttpException {
```

Comment:
Implement UpdateStatefulNodeState method for all state update ( pause/resume/recycle) since the endpoint is same. And also, as per the documenation, customer will look for UpdateStatefulNodeState method to update the state using the SDK.

Reply:
Incorporated. 
Implemented updateStatefulNodeState method which updates state - pause/resume/recycle (single endpoint).